devito/passes/iet/mpi.py

```diff
@@ -91,7 +91,7 @@ def rule1(dep, candidates):
 
             for f in hs.fmapper:
                 for n, i in enumerate(iters):
-                    candidates = set().union(*[i.dim._defines for i in iters[n:]])
+                    candidates = [i.dim._defines for i in iters[n:]]
```

Comment:
No need for set anymore?

Reply:
no, `._defines` is a set and the check now works at level of each `._defines` set, not the cumulative set